### PR TITLE
[Sample] Add a sample which converts a langchain app to flex flow with custom connection

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This pull request introduces a new feature to the `langchain-eval` example in the `flex-flows` directory. The new feature includes a workflow for evaluating language chains, as well as changes to the README, data, and requirements files. The most significant changes are:

1. A new GitHub workflow file `.github/workflows/samples_flexflows_langchaineval_langchaineval.yml` was added. This workflow is scheduled to run daily and on pull requests that modify specific paths. It sets up a Python 3.9 environment, installs requirements, and runs a notebook test, among other steps.

2. The `examples/README.md` file was updated to include a link to the `langchain-eval.ipynb` notebook and its corresponding workflow badge.

3. Two new data entries were added to the `examples/flex-flows/langchain-eval/data.jsonl` file.

4. The `examples/flex-flows/langchain-eval/flow.flex.yaml` file was added, which contains the schema and entry point for the new feature.

5. A new Python file `examples/flex-flows/langchain-eval/flow.py` was added. This file contains the `LangChainEvaluator` class, which is used to evaluate language chains.

6. The `examples/flex-flows/langchain-eval/langchain-eval.ipynb` notebook was added. This notebook provides a tutorial on how to use the new feature.

7. The `examples/flex-flows/langchain-eval/requirements.txt` file was updated to include the `promptflow` and `langchain` packages, as well as the `python-dotenv` package.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by elliotzh — APPROVED** [review]

File: examples/flex-flows/langchain-eval/requirements.txt
Comment (wangchao1230) [inline examples/flex-flows/langchain-eval/requirements.txt:1]: add readme for this sample

File: examples/flex-flows/langchain-eval/flow.py
Comment (wangchao1230) [inline examples/flex-flows/langchain-eval/flow.py:1]: eval-xx follow our naming pattern

File: examples/flex-flows/langchain-eval/flow.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+class LangChainEvaluator:
+    def __init__(self, custom_connection: CustomConnection):
+        self.custom_connection = custom_connection
+
+        # create llm according to the secrets in custom connection
+        if "anthropic_api_key" in self.custom_connection.secrets:
+            self.llm = ChatAnthropic(
+                temperature=0, anthropic_api_key=self.custom_connection.secrets["anthropic_api_key"]
```
Comment (wangchao1230) [inline examples/flex-flows/langchain-eval/flow.py:24]: what are the actual evaluate logic here？ reflect in th sample name

File: examples/flex-flows/eval-criteria-with-langchain/langchain-eval.ipynb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+ ],
+ "metadata": {
+  "build_doc": {
+   "author": [
+    "D-W-@github.com",
+    "wangchao1230@github.com"
+   ],
+   "category": "local",
+   "section": "Flow",
+   "weight": 10
```
[the remark this answers is not in the document]
Comment (D-W-) [reply examples/flex-flows/eval-criteria-with-langchain/langchain-eval.ipynb]: Updated to last order

**Review by brynn-code — APPROVED** [review]

File: examples/flex-flows/eval-criteria-with-langchain/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
+# Eval Conciseness Criteria with LangChain
+
+A example flow of converting LangChain criteria evaluator application to flex flow.
```
Comment (wangchao1230) [inline examples/flex-flows/eval-criteria-with-langchain/README.md:3]: what's a criteria  evaluator?

Comment (wangchao1230) [reply examples/flex-flows/eval-criteria-with-langchain/README.md:3]: may add some link here: https://python.langchain.com/docs/guides/productionization/evaluation/string/criteria_eval_chain/

**Review by wangchao1230 — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**